framework/web/UrlManager.php

```diff
@@ -431,22 +431,22 @@ public function createUrl($params)
             }
 
             if ($url !== false) {
-                if (strpos($url, '://') !== false) {
-                    if ($baseUrl !== '' && ($pos = strpos($url, '/', 8)) !== false) {
+                if (Url::isRelative($url)) {
+                    $url = ltrim($url, '/');
+                    return "$baseUrl/{$url}{$anchor}";
+                } elseif (Url::isProtocolAgnostic($url)) {
+                    if ($baseUrl !== '' && ($pos = strpos($url, '/', 2)) !== false) {
                         return substr($url, 0, $pos) . $baseUrl . substr($url, $pos) . $anchor;
                     }
 
                     return $url . $baseUrl . $anchor;
-                } elseif (strncmp($url, '//', 2) === 0) {
-                    if ($baseUrl !== '' && ($pos = strpos($url, '/', 2)) !== false) {
+                } else {
```

Comment:
No need for `else` - we could safely remove `else` and change `elseif` to `if`.

Reply:
It seems the whole block got duplicated; I'll have another look at this.